operator/apis/metrics/v1alpha1/keptnmetric_types.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// KeptnMetricSpec defines the desired state of KeptnMetric
+type KeptnMetricSpec struct {
+	// Source represents the provider name
+	Source string `json:"source"`
```

Comment:
if this can only be the provider name, maybe we should name it `SourceProviderName`?

Reply:
needs discussion, but sure why not, @thisthat ?

Reply:
as all PRs depend on this one, I would maybe leave it like this as we already align ourselves on this version